### PR TITLE
[DC-1305] Fix incorrect type mismatch flagging of nulls

### PR DESCRIPTION
**Pull request by jp3477**
This fixes incorrect flagging of null values as type mismatch in nullable columns.

### CODE REVIEW

File: omop_file_validator.py [hunk 1]
```diff
@@ -251,7 +249,7 @@ def run_checks(file_path, f):
                     submission_column_type = df[submission_column].dtype
 
                     # If all empty don't do type check
-                    if submission_column_type is not None:
```
Comment (jp3477) [inline omop_file_validator.py:287]: I'm not sure the dtype property of a series can be None.

File: omop_file_validator.py [hunk 2]
```diff
@@ -118,10 +118,8 @@ def find_error_in_file(column_name, cdm_column_type, submission_column_type,
 
         try:
             if i <= len(df) - 1:
-                if row[column_name]:
```
Comment (jp3477) [inline omop_file_validator.py:154]: NaN values were being accepted by this condition and compared via cast_type, which would often fail.

Comment (danyel-campbell) [reply omop_file_validator.py:154]: Nice catch!

File: omop_file_validator.py (hunk 1 above)
Comment (danyel-campbell) [reply omop_file_validator.py:287]: You're right, the dtype cannot be None.

[List of dtypes here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.dtypes.html)

File: omop_file_validator.py [hunk 3]
```diff
@@ -284,7 +282,7 @@ def run_checks(file_path, f):
                     submission_column_type = df[submission_column].dtype
 
                     # If all empty don't do type check
-                    if submission_column_type is not None:
+                    if not df[submission_column].isnull().values.all():
```
Comment (danyel-campbell) [inline omop_file_validator.py:285]: Nice change!

**Review by danyel-campbell — APPROVED** [review]
This looks good. Good catching the NaN values being accepted issue as well as the dtype being None.

File: omop_file_validator.py (hunk 2 above)
Comment (jp3477) [reply omop_file_validator.py:154]: Thanks!

**Pull request merged**